Draft for n512 configuration docs (#20)
ACCESS-NRI/access-am3-configs-doc: documentation/docs/pages/configs_experiments/n512e.md

# ACCESS-AM3 n512e configuration

> The `dev-n512e-aeroclim` configuration branches from `dev-96e` with a few changes to accommodate for the higher resolution requirements. The main differences with the `dev-n96e` beta release configuration are:

Reviewer: ```suggestion
The `n512e-aeroclim` configuration is based on the beta `n96e` configuration with a few changes to accommodate for the higher resolution requirements. The main changes in the `n512-aeroclim` configuration are:
```

When we officially beta release, we merge those dev branches into the respective release branches. The typical workflow is to branch individual changes off the `dev` branch, merge it back into the `dev` branch (and potentially cherry pick into other branches), then merge those dev branches back into the original `release` branch when we do a release.

Reviewer 2: Usually for the configuration, we use the full name of the branch so I would prefix all the names with `release-` in that text. I think it is less confusing to users if we use the full branch name even if strictly speaking it's the n96e and n512e-aeroclim configs.

Author: I didn't realise when I wrote this that the branch names were going to be different. Currently, "my" `dev-n512e-aeroclim` branch is going to be merged into `dev-n512e`. If I understand correctly, the plan is to then merge/create `release-n512e`?

> The typical workflow is to branch individual changes off the `dev` branch, merge it back into the `dev` branch (and potentially cherry pick into other branches), then merge those dev branches back into the original `release` branch when we do a release.

I think I need a map to follow this :joy:

Reviewer 2: The `dev-` to `release-` merge is so that we get a PR process for the final release. I actually wonder if the n512e branch should actually go directly into `release-` instead of access-am3-configs@access-nri:`dev-n512e`. But it's a bit late to make the decision. Let's get to this tomorrow!

Author: In any case, I'll edit the documentation to use `release-n512e`.

* Uses climatological aerosols instead of prognostic aerosols.
* The ancilliares uses high resolution data. 
* IO Server is activated to improve performance.

> * Specific list of output variables and frequencies (3- and 6-hourly). 

Reviewer: ```suggestion
* The ancillaries uses high resolution data. 
* IO Server is activated to improve performance.
* Modified list of output variables and frequencies (3- and 6-hourly). 
```

Can you add hyperlinks to the respective sections in this list?

The configuration uses a 1024x768 grid, with a nominal resolution of ~25 km and 85 vertical levels. 

![ACCESS-AM3 n512e configuration](images/orography_n96_n512.png)

## Initial conditions

The configuration can be initialise from a `n96` or `n512` restart. If there is a change of resolution, the reconfiguration step (`RECON`) will interpolate all the necessary variables. 

> The suite also include a `fix_cable_restart` step to populate CABLE variables after the reconfiguration. This is necessary when the model is initialise from low resolution initial conditions. 

Reviewer: ```suggestion
The suite also include a `fix_cable_restart` step to populate CABLE variables after the reconfiguration. This is necessary when the model is initialised from low resolution initial conditions. 
```

> The default configuration runs with a `n96` restart file from January 1, 2007. There may be other restart files available for different dates and resolutions, you can use the [AM3 subcategory](https://forum.access-hive.org.au/c/atmosphere/am3/243) on the Hive to get more information about this.  

Reviewer: ```suggestion
The default configuration runs with a `n96e` restart file from January 1, 2007. There may be other restart files available for different dates and resolutions. If you would like to find a restart for your particular purpose, please post in the [AM3 subcategory](https://forum.access-hive.org.au/c/atmosphere/am3/243) on the Hive Forum.  
```

Reviewer 2: Actually, I think the `e` should always be here because it denotes a different grid: moved by half a grid cell. The `e` stands for `endgame` which is a scheme introduced in the UM and that shifted the grid as well. 

So n96 ancillaries can't be used with a n96e configuration.

> To change the restart used edit the file `site/nci_gadi.rc`:

Reviewer: ```suggestion
To change the restart used, edit the file `site/nci_gadi.rc`:
```

```
{% set AINITIAL = '/path/to/restart/file' %}
```

The initial date and experiment length is defined in the `rose-suite.conf_nci_gadi` file. The default configuration runs for 1 month, from January 1, 2007. 

```
EXPT_BASIS='20070101T0000Z'
EXPT_RUNLEN='P1M'
```

## Ancilliaries

The ancillaries `n512` where generated using 2 different workflows. The suite `u-dj813` was adapted to create the ozone and ESA (sst and sea ice) ancillaries. All the other ancilliaries, including vegetation fraction, vegetation function and soil are generated with the CCI-Ancilliary-Suite and based on the 300m resolution [CCI Land Cover dataset](https://catalogue.ceda.ac.uk/uuid/c19b0914521144ab8c18c91d586c6847/).

Climatologies:

* `aerosols` (monthly mean)
* `soil` caracteristics

> * `veg.frac` and `and veg.func` (monthly mean)

Reviewer: I'm not sure whether we should describe these as climatological- they feel more like fixed in time than climatological. Is the purpose here to describe ancillaries that differ between the n96 and n512 configs?

Author: The idea behind this section was to highlight differences between n96e and n512e but also give people some extra information about inputs and forcings. I'll try to improve it tomorrow. I can remove the categories and pull some extra information from the READMEs.

Author: I added a few changes to this section, ready to review @Whyborn

* 3D ERA5 Diagnostics: 37 pressure levels, 6 hourly and monthly mean output
* 3D Standard Diagnostics: All model levels, 6 hourly and monthly mean output
* COSP
* 2D Extra: extra 2D variables, 3 hourly output
* 3D Extra: extra 3D variables, 3 hourly output

> Currently the COPS, 2D Extra and 3D Extra packages are not active because they haven't been fully tested. They can be activated from the GUI in `UM/namelist/Model Input and Output/STASH Request and Profiles/STASH request` (upper right corner drop-down menu).

Output files are saved in the UM native format in `share/data/History_Data/`. Some of the variables are saved in monthly files, while others are saved in daily files. Check [the spreadsheet](https://docs.google.com/spreadsheets/d/11S5hY54BPHzoi2v6KYZeUeVo-76Q8QLqwKfjJCPuCrI/edit?gid=1519343531#gid=1519343531) to see how the variables are saved. 

The output is then converted to netcdf format and saved in `share/data/History_Data/netCDF/`. 

To save space, there is an option to delete the field files after they are converted to netcdf. In `app/netcdf_conversion/rose-app.conf`, set:

```
REMOVE_FF=true
```

## Optimization

> The IO Server is active in the configuration to improve on walltime and SU usage. Usually the UM will read and write files on disk sequentially, but the IO Server gets extra processors to read and write files in parallel. This allows the UM to continue with the simulation while the IO Server is writing files to disk. 

Reviewer: ```suggestion
The IO Server is active in the configuration to reduce walltime and SU usage. Usually the UM will read and write files on disk sequentially, but the IO Server gets extra processors to read and write files in parallel. This allows the UM to continue with the simulation while the IO Server is writing files to disk. 
```
